build-system/npm-publish/write-package-files.js

```diff
@@ -42,6 +58,22 @@ async function writePackageJson() {
     return;
   }
 
+  const exports = {
+    '.': './preact',
+    './preact': {
+      import: './dist/component-preact.module.js',
+      require: './dist/component-preact.js',
+    },
+    './react': {
+      import: './dist/component-react.module.js',
+      require: './dist/component-react.js',
+    },
+  };
+
+  if (await hasStylesheets()) {
+    exports['./css/*'] = './dist/*';
```

Comment:
Why not add an individual export for each css file? We're already doing the glob, so we've already paid for the filesystem access.

Reply:
Great question!

Functionality-wise both would be equal—at least I couldn't find any difference in my testing. So it's more a matter of preference I suppose.

[The docs](https://nodejs.org/api/packages.html#packages_subpath_patterns) say this though:

> For packages with a small number of exports or imports, we recommend explicitly listing each exports subpath entry. But for packages that have large numbers of subpaths, this might cause `package.json` bloat and maintenance issues.

Reply:
My only issue is that this exposes everything in the `dist` directory, eg `"css/component-react.module.js"` is now valid.

Reply:
That's a good point.

I originally had this as `exports['./css/*'] = './dist/*.css';`, but that would require imports like `import '@ampproject/amp-base-carousel/css/styles';`, which is confusing due to the lack of file extension not making it 100% clear one is actually importing a CSS file.

So yeah, individual exports would avoid this.